Fix #10250 - remove manual class discriminator configuration

mattattui opened the pull request:
As discussed in #10250, the documentation refers to functionality that sadly does not exist ([yet!](https://github.com/symfony/symfony/pull/28351))
To fix this:

* I've removed the reference to `addClassMap()`
* Rearranged the section slightly to reflect that the `ClassDiscriminatorFromClassMetadata` resolver is already set up when using the Symfony framework.
* Split the example into separate setup and example usage blocks.

I think this is the first time I've done more to the symfony docs than fix a typo, so I'm sorry and please be gentle with me if I've messed anything up :)

mattattui committed:
Remove sadly non-existent feature from example

mattattui committed:
Rearrange configuration and example

Setting up the `ClassDiscriminatorFromClassMetada` resolver is unnecessary when using the framework, so I've made that clear.

I've also separated the setup from the example usage so that the example discriminator map configuration can be shown first.

javiereguiluz reviewed (approved):
@inanimatt thanks for this contribution! You did it perfectly.

The only minor change needed is that we no longer talk about "Symfony Standard Edition" in Symfony 4 docs, because Flex usage is assumed.

So, we should reword this:

```
When using the Symfony Standard Edition ...
```

by something like this:

```
When using this component inside a Symfony application ...
```

mattattui committed:
Fix capitalisation and standard edition references

I've applied consistent capitalisation to "Serializer component", and replaced references to the Symfony Standard Edition.

javiereguiluz commented:
@inanimatt thanks a lot for contributing these fixes! It was merged in https://github.com/symfony/symfony-docs/commit/ee8fa8e6c678e8ee3a712c9c37e59ea9d1991ae1

sroze commented:
Thank you @inanimatt.
